src/App.tsx

```diff
@@ -1,61 +1,188 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './Types/user';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import { Todo } from './Types/todo';
+import { FormFields } from './Types/formFields';
+
+const formFields: FormFields = {
+  title: '',
+  user: '0',
+};
+
+const findUserByUserId = (userId: number): User | null => {
+  return usersFromServer.find((user: User) => user.id === userId) || null;
+};
+
+const makeNewTodo = (
+  id: number,
+  title: string,
+  userId: number,
+  user: User | null,
+  completed: boolean = false,
+) => {
+  return {
+    id: id,
+    title: title,
+    completed: completed,
+    userId: userId,
+    user: user,
+  };
+};
+
+const preparedTodos = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: findUserByUserId(todo.userId),
+  };
+});
+
+const formFieldsErrors = () => {
+  const fieldsErrors: { [key: string]: boolean } = {};
+
+  for (const key in formFields) {
+    fieldsErrors[key] = false;
+  }
+
+  return fieldsErrors;
+};
 
 export const App = () => {
+  const [todoList, setTodoList] = useState<Todo[]>(preparedTodos);
+  const [errorsInForm, setErrorsInForm] = useState(formFieldsErrors());
+  const [newTodoForm, setNewTodoForm] = useState(formFields);
+  const [isTouched, setIsTouched] = useState(false);
+
+  const handleValidation = (newTodo: FormFields) => {
+    for (const key in newTodo) {
+      const isError = newTodo[key] === formFields[key];
+
+      if (isError) {
+        setErrorsInForm(prevState => {
+          return {
+            ...prevState,
+            [key]: true,
+          };
+        });
+      }
+    }
+  };
+
+  const handleErrorAppear = (isIncorrectField: boolean) => {
```

Comment:
`handle` or `check`?
I think ur function doesn't do anything, so this will be more descriptive
```suggestion
  const checkErrorAppearance = (isIncorrectField: boolean) => {
```